File: backend/src/models/Comments.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export interface IComments extends Document {
    userId: number,
    articleId: Types.ObjectId,
    content: string,
    createTime: Date,
    deleted: boolean,
    reviewStatus: 'APPROVED' | 'REJECTED'
}

const CommentsSchema: Schema = new Schema({
    userId: {
        type: Number,
        require: true
    },
    articleId: {
        type: Schema.Types.ObjectId,
        ref: 'Article',
        require: true
    },
    content: {
        type: String,
        require: true,
        trim: true
    },
    createTime: {
        type: Date,
        default: Date.now()
    },
    reviewStatus: {
        type: String,
        enum: ['REJECTED', 'APPROVED'],
        default: 'REJECTED'
    },
    deleted: {
        type: Boolean,
        default: false
    }
}, {
    timestamps: {
        createdAt: 'createTime',
        updatedAt: 'updateTime'
    }
})

CommentsSchema.index({ articleId: 1, reviewStatus: 1, deleted: 1 });
CommentsSchema.index({ userId: 1, createTime: -1 });

const Comments = mongoose.model<IComments>('articles_comments', CommentsSchema);
export default Comments;

File: backend/java-backend/src/main/java/com/example/demo/model/mongo/Comments.java
package com.example.demo.model.mongo;

import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "articles_comments")
public class Comments {

    @Id
    private String id;

    private Integer userId;
    // articleId 字段类型为 ObjectId，与查询类型保持一致，避免 MongoDB 类型不匹配
    private ObjectId articleId;
    private String content;
    private LocalDateTime createTime;
    private String reviewStatus;  // APPROVED, REJECTED
    private Boolean deleted;
}
